### PR TITLE
EVG-14603: randomly use an alt github token

### PR DESCRIPTION
**Pull request by hhoke**

### CODE REVIEW

**Comment by hhoke** [comment]
confirmed the random alt token selection works on staging

File: config.go [hunk 1]
```diff
@@ -551,31 +551,60 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
-func (s *Settings) GetGithubOauthString() (string, error) {
+func (s *Settings) GetGithubOauthString() ([]string, error) {
```
Comment (ablack12) [inline config.go:554]: nit: "strings" now right

File: config.go [hunk 2]
```diff
@@ -551,31 +551,60 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
...
 	if err != nil {
 		return "", err
 	}
+	timeSeed := time.Now().Nanosecond()
+	for i, oauthString := range oauthStrings {
+		randomOffset := timeSeed + i
+		if randomOffset%len(oauthStrings) == 0 {
+			splitToken, err := splitToken(oauthString)
+			if err != nil {
+				err = errors.Wrapf(err, "problem with github_alt#%d", i)
```
Comment (ablack12) [inline config.go:593]: I think we should drop the # since that doesn't match the actual token name

File: config.go [hunk 3]
```diff
@@ -551,31 +551,60 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
...
 		return "", errors.New("not defined")
 	}
 
-	oauthString, err := s.GetGithubOauthString()
+	oauthStrings, err := s.GetGithubOauthString()
 	if err != nil {
 		return "", err
 	}
+	timeSeed := time.Now().Nanosecond()
+	for i, oauthString := range oauthStrings {
```
Comment (ablack12) [inline config.go:588]: I get why we're randomizing kind of but the process of it is a bit confusing to me; is it possible we could just get stuck in like an infinite loop? And if there's an error with one of the oauth strings, should we just try another one?

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: config.go (hunk 3 above)
Comment (hhoke) [reply config.go:588]: The modulo arithmetic works out from first principles, and I ran through line-by-line with a couple examples. We are guaranteed to choose one and only one token per instance of `oauthStrings`. 

There are n distinct modulo equivalence classes for modulus n. For example, there are two modulo equivalence classes for mod 2: {0,2,4...} and {1,3,5...}. In any loop, we advance through n consecutive integers, where n is the length of oauthStrings. Each time we iterate, we enter a new equivalence class. So, by something like the pigeonhole principle, we have to land on the equivalence class of {0,n,2n,3n...} once and only once. 
Say we get a `timeSeed` that is in the equivalence class with n+1. Then, on the final iteration, `randomOffset = (n+1) + (n-1)`, or 2n. 2n is in the equivalence class with 0.

Comment (hhoke) [reply config.go:588]: I like the idea of just trying another one if there's an error. I'll code that up

Comment (ablack12) [reply config.go]: Thank you for that walkthrough!

File: config.go [hunk 4]
```diff
@@ -551,31 +551,61 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
...
 		return "", err
 	}
+	timeSeed := time.Now().Nanosecond()
+	randomStartIdx := timeSeed % len(oauthStrings)
+	var oauthString string
+	for i := range oauthStrings {
+		oauthString = oauthStrings[randomStartIdx+i]
+		splitToken, err := splitToken(oauthString)
+		if err != nil {
+			grip.Error(errors.Wrapf(err, "problem with github_alt%d", i))
```
Comment (ablack12) [inline config.go:594]: In practice, logging errors to grip is unwieldy / unnecessary if we don't include some message fields to keep it organized.

**Review by ablack12 — APPROVED** [review]
LGTM with suggestions

**Pull request merged**